dxp/demystifying_aui_script_and_liferay-frontend_component_tags.md

```diff
@@ -0,0 +1,114 @@
+
+```jsp
+<aui:script require="metal-clipboard/src/Clipboard">
+    new metalClipboardSrcClipboard.default();
+</aui:script>
+```
+
+Also, we can set an alias with the `as` word after the provided path. It will be looking like:
+
+```jsp
+<aui:script require="metal-dom/all/dom.js as dom">
+
+	function myFunction(element) {
+		dom.addClasses(element, 'hide');
+	}
+
+	Liferay.once('something', myFunction);
+
+</aui:script>
+```
+
+#### Settling the way which the script will be placed on the page using `position`:
+
+This property will change the way in which / how the script will be injected into the page. By default, those scripts that are wrapped by `aui:script` will be added to the minified files as described here (Pure JS on the JSP [PUT THE LINK HERE OH]).
+
+When placing the option of `inline` for this property, the scripts will be placed in a script tag obeying the order of the DOM where the JSP is being rendered. See [here](https://github.com/liferay/liferay-portal/blob/master/util-taglib/src/com/liferay/taglib/aui/ScriptTag.java#L143..L165).
+
+#### `sandbox`:
+
+The purpose for the sandbox attribute is to wrap the contents of the script tag in a self-invoking function, and to enable short-handing the `$` and `_` variables rather than calling `AUI.$` and `AUI._`.
+
+#### Caveats:
+
+We couldn’t use both `require` and `use` in the same file. Because `use` property calls the modularization for alloy-ui modules and `require` follows ES6 modularized files.
+
+You can see more information about `aui:script` [here](https://help.liferay.com/hc/en-us/articles/360017882752-Loading-Modules-with-AUI-Script-in-Liferay-DXP).
+
+## `liferay-frontend:component` tag
+
+We can use this tag for removing JavaScript code from JSPs. This tag accepts a `module` property wether we can pass a relative path for the script to be loaded. Also, this tag registers automatically the component to [Liferay’s component global registry](https://github.com/liferay/liferay-frontend-guidelines/blob/master/dxp/liferay_component.md#register), depending on the `componentId` property. When using this tag, `namespace`(or `portletNamespace`) and `spritemap` props will turn available.
+
+Here is an example of usage of this tag:
+
+On the JSP context:
+
+```jsp
+
+...
+<liferay-frontend:component
+	componentId="myComponent"
+	module="path/to/my/js/myComponent.js"
+/>
+...
+```
+
+And when exporting the `myComponent.js` file we can define a default function there like:
+
+```js
+export default function myComponent(props) {
+	console.log(props.namespace);
+	// it will return the portlet namespace where the liferay-frontend:component tag were called
+	console.log(props.spritemap);
+	// it will return the path to the page's spritemap
+}
+```
+
+This taglib can be very handy if you want to use modern ES6 features, since We don’t have support for transpiling ES6 from taglibs.
```

Comment:
taglib -> tag

Reply:
- "ES6" → just say "modern JS" (because otherwise you're making it sound like we can only use the already-ancient ES2015, which we've already moved far beyond, and we transpile a bunch of ES2019 stuff out-of-the-box in DXP).
- "We" → "we"

It's not so much "we don't have support for transpiling ES6 from taglibs" though, is it? It's more like "we don't transpile JS found in JSP files" — is that right?

Reply:
Makes sense replacing "ES6" occurrences with "modern JS" for improve the readability of non-javascript persons :p

Reply:
To be honest, I don't think any non-JS people are going to be reading this. My main concern is that we don't use a term that refers to a standard we don't use any more.